lib/kube/proxy/auth.go

```diff
@@ -140,13 +142,49 @@ func (f *Forwarder) getKubeDetails(ctx context.Context) error {
 			})
 		if err != nil {
 			f.log.WithError(err).Warnf("Failed to create cluster details for cluster %q.", cluster)
-			return trace.Wrap(err, "setting up details for cluster %q", cluster)
+			// If the cluster is not available, spawn a goroutine that will retry
+			// until the cluster is available.
+			go f.retryClusterRegistrationUntilAvailable(ctx, kubeCluster, clusterCreds)
```

Comment:
This is good for a quick fix, but since the retry function duplicates the responsibility of creating cluster details object and registering the cluster, please consider pulling out the logic into a common place. This will reduce the probability that someone will modify one piece and not another.